third_party/amd/lib/TritonAMDGPUTransforms/OptimizeDotOperands.cpp

```diff
@@ -205,6 +205,69 @@ class ReuseShmemForDirectAndTransposedUse : public OpRewritePattern<LoadOp> {
   triton::AMD::ISAFamily isaFamily;
 };
 
+template <typename OpTy>
+class AllocSharedMemForUpcastedScales : public OpRewritePattern<OpTy> {
+public:
+  using OpRewritePattern<OpTy>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(OpTy op,
+                                PatternRewriter &rewriter) const override {
+    auto forOp = op->template getParentOfType<scf::ForOp>();
+    if (!forOp)
+      return rewriter.notifyMatchFailure(op,
+                                         "Don't alloc lds outside for loop");
+
```

Comment:
Do we also want to enable this pattern on mi300? Right now given we aren't using `amdgpu::ScaledUpcastFp*Op` op there yet so it won't kick in. But wondering whether we want to check isaFamily here to exclude mi300 from the get-go given the limited LDS there..